src/app/components/Map/Map.tsx

```diff
@@ -203,44 +226,72 @@ export default function MapBox(): JSX.Element {
   }
 
   function onClear() {
+    setLocation1(null)
+    setLocation2(null)
     setLocationName1('')
     setLocationName2('')
   }
   function showMap() {
-    setShowMapPage(!showMapPage)
-    console.log(showMapPage)
+    if (!showMapPage && locationName1 && locationName2) {
+      setShowMapPage(!showMapPage)
+    } else if (showMapPage) {
+      setShowMapPage(!showMapPage)
+      setLocation1(null)
+      setLocation2(null)
+      setLocationName1('')
+      setLocationName2('')
+    } else {
+      alert('please set both inputs')
+    }
   }
 
-  getRoute(location1, location2)
+  location1 && location2 ? getRoute(location1, location2) : null
+
+  console.log(location1, location2)
 
   return (
     <>
-      <InputContainer>
-        <h1>See U There</h1>
+      <InputPage hidden={showMapPage}>
```

Comment:
Maybe `InputSection` would be a bit more fitting, since this is part of a page right?